vm/src/obj/objdict.rs

```diff
@@ -206,11 +206,10 @@ impl PyDictRef {
         if let Some(value) = self.entries.borrow().get(vm, &key)? {
             return Ok(value);
         }
-
-        if let Ok(method) = vm.get_method(self.clone().into_object(), "__missing__") {
+        if let Some(method_or_err) = vm.get_method(self.clone().into_object(), "__missing__") {
```

Comment:
I think it is probably not handy to return `Option<Result<..>>`. Why not reverse the two so that we have the error as the outer type? Like this: `Result<Option<..>>`. Then we can use the `?` operator more intuitively.

Reply:
I did it because `Option` is this case means "whether a method present on the object, or not". And if it's present, then we can check whether `__get__` method raised exception. 

There are three possible states
1. No method, `get_method` returns `None`.
2. Method exists, but `__get__` raised exception, returns `Err(some_error)`.
3. No exception, return method. 

Moving `Option` inside converts it into 4 states: 
1/2. Whether exception was returned from `__get__` or not. 
3/4. Whether method exists or not. 
It doesn't compatible with reality of the `get_method`.